source/server/server.cc

```diff
@@ -147,6 +147,14 @@ int InstanceImpl::getListenSocketFd(const std::string& address) {
   return -1;
 }
 
+Network::ListenSocket* InstanceImpl::getListenSocketByIndex(uint32_t index) {
+  if (index < config_->listeners().size()) {
```

Comment:
Why's it faster to iterate over the list rather than advance an iterator? AFAIK, you can't randomly access an element in `std::list`, so you have to do something O(n).